src/analysis_utils.py

```diff
@@ -0,0 +1,316 @@
+"""
+This module contains all the functions responsible for performing the analysis in this
+project.
+"""
+
+import pathlib
+
+import pandas as pd
+from copairs import map
+
+from .data_utils import label_control_types, split_meta_and_features
+from .io_utils import load_config
+
+
+def calculate_dmso_map_batch_profiles(
+    batched_profiles: dict,
+    configs: str | pathlib.Path | dict,
+    outdir_path: str | pathlib.Path,
+    shuffled: bool | None = False,
+) -> None:
+    """ Calculate average precision (AP) and mean average precision (mAP) scores
+    for DMSO-treated profiles across batches.
```

Comment:
```suggestion
    """ Calculate average precision (AP) and mean average precision (mAP) scores
    for DMSO-treated profiles across batches of replicate plates.
```
Might be more descriptive, but super optional.